ftw/solr/browser/maintenance.py

```diff
@@ -17,13 +20,47 @@
 from zope.component import queryUtility
 from zope.component.hooks import getSite
 import logging
+from logging.handlers import TimedRotatingFileHandler
 import transaction
 
 
 logger = getLogger('ftw.solr.maintenance')
 logger.setLevel(logging.DEBUG)
 
 
+def get_logfile_path():
+    """Determine the path for solr-maintenance.log.
+    This will be derived from Zope2's EventLog location, in order to not
+    have to figure out the path to var/log/ and the instance name ourselves.
+    """
+    zconf = getConfiguration()
+    eventlog = getattr(zconf, 'eventlog', None)
+    if eventlog is None:
+        return None
+
+    handler_factories = eventlog.handler_factories
+    eventlog_path = handler_factories[0].section.path
+    logdir = dirname(eventlog_path)
+    return pjoin(logdir, 'solr-maintenance.log')
+
+
+def setup_maintenance_loghandler():
+    logfile_path = get_logfile_path()
+    if not logfile_path:
+        return
+
+    handler = TimedRotatingFileHandler(
+        logfile_path,
+        when='midnight', backupCount=7)
+    handler.setLevel(logging.DEBUG)
```

Comment:
wouldn't `INFO` suffice here? can be we we don't log much, but theoretically `DEBUG` can be quite verbose.

Reply:
It's just the same level as the `ftw.solr.maintenance` logger defined above, to which the handler is getting attached. For now this level isn't used anyway from what I can tell, but this way we at least make sure to not drop any messages.